### DISCUSSION TITLE
Technology Review [Due Nov 14 Tue]

### DISCUSSION THREAD

**Discussion by LilacHo** [UndefiedCse/Heat-Island#15] [General]
5-7 mins; Recommended: ~1 min Background (don't be long on background)
Emphasize on one package with ~ 3 pieces

Refer to [Slide](https://github.com/UWSEDS/lecture-materials/raw/master/communication.pptx) on the website as well as the beginning part of Nov.7 class recording for more details.

Structure:
- Background: your application and why you want to use it, 1 slide
- What technologies you considered: 1 slide
- Choice: what you chose to use, summary  of how it works, 1 slide
- Appeal of choice: 1 slide
- Drawbacks of choice: 1 slide

Some other comments: 
For ONE purpose (decision) of our project, evaluate 3 different packages that can help.
Maybe we can do it for visualization? idk what we want to do.

Need 1-2 minutes of background.

A good (active) package on github has multiple forked and stars
Has good open:closed issues ratio
Should have a good README, explaining documentation + installation guide

Should have slides!

**Comment by LilacHo** [comment]
Packages for map visualization can be commented here

**Reply by loriwon-uw** [reply]
1) Folium: Looks like it is python accessible, visualize in a "Leaflet map", idk what that means.
Likely a little fancy, but can display map based on latitude/longitude.
Can control the amount of zoom + max latitude/longitude, to limit the bounds of Seattle if using seattle map, or Edinborough for those weather stations.

https://python-visualization.github.io/folium/latest/

**Reply by LilacHo** [reply]
**Bokeh** is a Python interactive visualization library that targets modern web browsers for presentation. It is good for creating interactive and real-time streaming maps. 
I think the interactive part can be useful since we want the map to respond our request like selecting a building.

https://github.com/bokeh/bokeh

**Reply by krittin-kulrat** [reply]
According to [Interactive large plot with ~20 million sample points and gigabytes of data](https://stackoverflow.com/questions/5854515/interactive-large-plot-with-20-million-sample-points-and-gigabytes-of-data) in Stackoverflow. One of answers compared each package

*Note that this question was ask 12 years ago*
 
**Mayavi** is a general purpose 3D scientific data visualization and plotting in Python. *There is an interactive application and I don't know whether we need the program to visualize the data.* There is example in the [Documentation](https://docs.enthought.com/mayavi/mayavi/index.html) about [plotting radar data from the NASA](https://docs.enthought.com/mayavi/mayavi/auto/example_canyon.html#example-canyon).

[GitHub repo](https://github.com/enthought/mayavi)
 
Pro:
- Able to visualize big data
- Open source package
- Highly interactive (According to the documentation)
- Simple and clean scripting interface
- Can save rendered visualization

Con:
- Dependencies are rather heavy (According to the documentation, and might be conflicted with `rasterio`)
- Latest release was 2 years ago (but last merge was 3 months ago)
- Not very active in the [repo](https://github.com/enthought/mayavi)

**Comment by loriwon-uw** [comment]
Goals for our map display:
1) Initially display the general city of Edinborough
2) After coordinate input, zoom into region + overlay with heat or height map
3) Toggle between heat/height color map
4) Scale bar, legend
5) Minimal storage/power required as possible, especially with larger datasets

**Comment by loriwon-uw** [comment]
Mayavi 
- Requires map import before display. Requires manual selection of region of interest. Identified using its own ID.
- Zoom is intuitive
- Naturally is a topographical map; likely can adjust color based on temperature (simultaneous display)
- no scale, legend tbd
- power/performance can be optimized.
- Old comparison says this is great.
- Big-data compatible.
- Easy to code/use. Open source.

**Comment by loriwon-uw** [comment]
Bokeh
- Can choose initial display city. Zoom easy.
- Color overlay with temperature/height seems possible
- Requires new image/page for displaying temperature and height.
- Legend yes. Scale bar no.
- Probably can handle larger datasets.
- Good UI, new, points clickable.
- Web browser

**Comment by LilacHo** [comment]
Folium
- yes if set up coordinate
- zoom is interactive 
- heat map and heat map with time
- Layer is ok
- No scale 
- Have a slider 
- if import from other package, yes legend
- Doesn't seem small, react slow-ish

**Comment by krittin-kulrat** [comment]
[Google Side for presentation](https://docs.google.com/presentation/d/1o48f3xMKLNGfdCzvENGFqi-c3PfGFukITIH8jfzuYI0/edit?usp=sharing)

**Reply by LilacHo** [reply]
@JornsenChao

**Reply by JornsenChao** [reply]
My preference: 1 folium 2 Bokeh 3 Mayavi.
Vote for folium
